# Fase 1: Análisis Exploratorio de Datos(EDA).

Antes de llevar a cabo el proyecto es crucial comprender mejor el conjunto de datos y sus características. Para ello deberás hacer un análisis exploratorio detallado del conjunto de datos para familiarizarte con ellos y entender que información tenemos.

# 📊 Preguntas clave para el análisis de rotación

| Nº | Pregunta | Variables involucradas | Objetivo del análisis |
|----|----------|------------------------|------------------------|
| 1 | ¿Qué porcentaje de empleados abandona la empresa? | Attrition | Medir la rotación general. |
| 2 | ¿La rotación cambia según departamento? | Department, Attrition | Comparar rotación por áreas. |
| 3 | ¿Qué puestos tienen más riesgo de rotación? | JobRole, Attrition | Identificar roles con mayor salida. |
| 4 | ¿La satisfacción laboral influye en la salida? | JobSatisfaction, EnvironmentSatisfaction, Attrition | Evaluar impacto de la satisfacción. |
| 5 | ¿Las horas extra están relacionadas con mayor rotación? | OverTime, Attrition | Ver si trabajar más horas aumenta la salida. |
| 6 | ¿La distancia desde casa afecta a la retención? | DistanceFromHome, Attrition | Analizar si vivir lejos influye en irse. |
| 7 | ¿El salario mensual influye en la rotación? | MonthlyIncome, Attrition | Comparar ingresos entre quienes se quedan y se van. |
| 8 | ¿La antigüedad reduce o aumenta la rotación? | YearsAtCompany, TotalWorkingYears, YearsInCurrentRole | Ver cómo influye la experiencia y permanencia. |
| 9 | ¿La falta de promoción está relacionada con la salida? | YearsSinceLastPromotion, Attrition | Evaluar si no promocionar aumenta la rotación. |
| 10 | ¿Qué variables necesitan limpieza antes del análisis? | EmployeeCount, Over18, StandardHours + nulos/duplicados | Identificar columnas constantes y problemas de calidad. |


añadir pregunta de género a esta tabla?

In [33]:
import pandas as pd
import numpy as np #Esto es para trabajar con arrays y operaciones matemáticas
import warnings #Esto sirve para manejar las advertencias que puedan surgir durante la ejecución del código
warnings.filterwarnings('ignore')

# Mostrar todas las columnas sin truncar
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', 60)

print("✅ Librerías cargadas correctamente")


✅ Librerías cargadas correctamente


In [3]:
df = pd.read_csv('hr.csv')

print(f"El número de filas y columnas es: {df.shape[0]} filas y {df.shape[1]} columnas")

El número de filas y columnas es: 1474 filas y 35 columnas


In [ ]:
df.head(5)

,Age,Attrition,BusinessTravel,DailyRate,Department,DistanceFromHome,Education,EducationField,EmployeeCount,EmployeeNumber,EnvironmentSatisfaction,Gender,HourlyRate,JobInvolvement,JobLevel,JobRole,JobSatisfaction,MaritalStatus,MonthlyIncome,MonthlyRate,NumCompaniesWorked,Over18,OverTime,PercentSalaryHike,PerformanceRating,RelationshipSatisfaction,StandardHours,StockOptionLevel,TotalWorkingYears,TrainingTimesLastYear,WorkLifeBalance,YearsAtCompany,YearsInCurrentRole,YearsSinceLastPromotion,YearsWithCurrManager
0,41.0,Yes,Travel_Rarely,1102,Sales,1,2,Life Sciences,1,1,2,Female,94,3,2,sALES eXECUTIVE,4.0,Single,5993.0,19479,8,Y,Yes,11,3,1,80.0,0,8,0.0,1,6,4,0,5.0
1,49.0,No,Travel_Frequently,279,Research & Development,8,1,Life Sciences,1,2,3,Male,61,2,2,rESEARCH sCIENTIST,2.0,Married,5130.0,24907,1,Y,No,23,4,4,NaN,1,10,3.0,3,10,7,1,7.0
2,37.0,Yes,Travel_Rarely,1373,Research & Development,2,2,Other,1,4,4,Male,92,2,1,lABORATORY tECHNICIAN,3.0,Single,2090.0,2396,6,Y,Yes,15,3,2,NaN,0,7,3.0,3,0,0,0,0.0
3,33.0,No,Travel_Frequently,1392,Research & Development,3,4,Life Sciences,1,5,4,Female,56,3,1,rESEARCH sCIENTIST,3.0,Married,2909.0,23159,1,Y,Yes,11,3,3,80.0,0,8,3.0,3,8,7,3,0.0
4,27.0,No,Travel_Rarely,591,Research & Development,2,1,Medical,1,7,1,Male,40,3,1,lABORATORY tECHNICIAN,2.0,Married,3468.0,16632,9,Y,No,12,3,4,80.0,1,6,3.0,3,2,2,2,2.0


Cositas para cambiar:
- JobRole --> mal escrito
- MaritalStatus --> en alguno pone marreid en vez de married y nulos
- EducationField --> Error de escritura, espacios... arreglar con .strip()
- AGE está en formato float, tenemos que convertirla a int

Eliminar:
- EmployeeCount, Over18, StandardHours porque no aportan nada a nuestro análisis y ya veremos si tambien eliminamos EmployeeNumber (un número único para cada empleado, no tiene ningún valor predictivo).

Estudiar:
- Attrition (DESGASTE) es la columna que nos interesa analizar, es la variable objetivo, es decir, lo que queremos predecir. Es una variable categórica, tiene dos valores: Yes y No. Tenemos que convertirla a 0 y 1 para poder trabajar con ella.

**¿Qué observamos?**
- Hay columnas de datos personales (`Age`, `Gender`, `MaritalStatus`)
- Columnas laborales (`Department`, `JobRole`, `YearsAtCompany`)
- Columnas de satisfacción y rendimiento (`JobSatisfaction`, `PerformanceRating`, `WorkLifeBalance`)
- La variable objetivo: `Attrition` — si el empleado se fue (Yes) o se quedó (No)
- Columna `JobRole` con un formato de texto extraño (mayúsculas y minúsculas mezcladas)

In [5]:
df.info

<bound method DataFrame.info of        Age Attrition     BusinessTravel  DailyRate              Department  \
0    41.00       Yes      Travel_Rarely       1102                   Sales   
1    49.00        No  Travel_Frequently        279  Research & Development   
2    37.00       Yes      Travel_Rarely       1373  Research & Development   
3    33.00        No  Travel_Frequently       1392  Research & Development   
4    27.00        No      Travel_Rarely        591  Research & Development   
...    ...       ...                ...        ...                     ...   
1469 34.00        No      Travel_Rarely        628  Research & Development   
1470 28.00        No      Travel_Rarely        866                   Sales   
1471 53.00        No      Travel_Rarely       1084  Research & Development   
1472 24.00       Yes      Travel_Rarely        240         Human Resources   
1473 45.00        No      Travel_Rarely       1339  Research & Development   

      DistanceFromHome  Educati

**Problemas detectados:**

🔴 **Columnas con nulos** (Non-Null Count < 1.474):
| Columna | Nulos | Impacto para el proyecto |
|---|---|---|
| `YearsWithCurrManager` | 148 | Dato laboral importante |
| `MaritalStatus` | 132 | Dato personal |
| `BusinessTravel` | 117 | Dato laboral |
| `TrainingTimesLastYear` | 88 | Clave para el perfil candidato |
| `Age` | 73 | Dato básico |
| `EducationField` | 58 | Contexto del empleado |
| `OverTime` | 44 | Clave para el perfil candidato |
| `JobSatisfaction` | 29 | Clave para el perfil candidato |
| `Department` | 29 | Dato laboral |
| `MonthlyIncome` | 14 | Dato económico |

🔴 **Tipos incorrectos:**
- `Age`, `JobSatisfaction`, `MonthlyIncome`, `TrainingTimesLastYear`, `YearsWithCurrManager` aparecen como `float64` cuando deberían ser `int` — nadie tiene 32.0 años ni 3.0 niveles de satisfacción. Esto ocurre porque pandas convierte a float las columnas con nulos.

→ Todo esto se corregirá en la **Fase 2**.

In [8]:
## Análisis detallado de valores nulos

#Calculamos exactamente cuántos nulos hay en cada columna y qué porcentaje representan sobre el total. Un porcentaje alto de nulos puede indicar que esa columna no es fiable.

nulos = df.isnull().sum()
pct   = (nulos / len(df) * 100).round(2)

resumen_nulos = pd.DataFrame({
    'Nulos': nulos,
    'Porcentaje (%)': pct
}).sort_values('Nulos', ascending=False)

print("Columnas CON nulos:")
print(resumen_nulos[resumen_nulos['Nulos'] > 0].to_string())
print(f"\nColumnas SIN nulos: {(nulos == 0).sum()}")
print(f"Total de valores nulos en el dataset: {nulos.sum()}")

Columnas CON nulos:
                       Nulos  Porcentaje (%)
StandardHours            164           11.13
YearsWithCurrManager     148           10.04
MaritalStatus            132            8.96
BusinessTravel           117            7.94
TrainingTimesLastYear     88            5.97
Age                       73            4.95
EducationField            58            3.93
OverTime                  44            2.99
JobSatisfaction           29            1.97
Department                29            1.97
MonthlyIncome             14            0.95

Columnas SIN nulos: 24
Total de valores nulos en el dataset: 896


**Interpretación:**
- Ninguna columna supera el 12% de nulos, así que no hay que descartar ninguna.
- `StandardHours` tiene nulos pero como es una columna redundante se eliminará igualmente.
- Los nulos se imputarán en la Fase 2: **mediana** para columnas numéricas (más robusta que la media ante valores extremos) y **moda** para columnas categóricas (el valor más frecuente).


## Pregunta 1 - ¿Qué porcentaje de empleados abandona la empresa? 

`Attrition` es la columna más importante del dataset. Indica si el empleado abandonó la empresa (`Yes`) o sigue en ella (`No`). 

Para el proyecto de cursos de idiomas, **solo nos interesan los empleados con `Attrition = No`**. Ofrecer un curso caro a alguien que ya se fue o que tiene alta probabilidad de irse sería un desperdicio de recursos.

In [10]:
print("=== DISTRIBUCIÓN DE ATTRITION ===\n")
print(df['Attrition'].value_counts())
print()
print("Porcentaje:")
print((df['Attrition'].value_counts(normalize=True) * 100).round(1).astype(str) + '%')
print()

# Universo de candidatos potenciales
candidatos = df[df['Attrition'] == 'No']
print(f"━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━")
print(f"Total empleados en el dataset:       {len(df):>5}")
print(f"Empleados que se fueron (excluidos): {df['Attrition'].eq('Yes').sum():>5}")
print(f"Candidatos potenciales al curso:     {len(candidatos):>5}")
print(f"━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━")


=== DISTRIBUCIÓN DE ATTRITION ===

Attrition
No     1236
Yes     238
Name: count, dtype: int64

Porcentaje:
Attrition
No     83.9%
Yes    16.1%
Name: proportion, dtype: str

━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
Total empleados en el dataset:        1474
Empleados que se fueron (excluidos):   238
Candidatos potenciales al curso:      1236
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━


**Interpretación:**
- El **16,1%** de los empleados se fue de la empresa — esto es la tasa de rotación.
- El **83,9%** sigue en la empresa → **1.236 candidatos potenciales** antes de aplicar cualquier otro filtro.
- A partir de aquí, todos los análisis de perfil se harán sobre este subconjunto.

## PREGUNTA 2. LA ROTACION CAMBIA SEGÚN DEPARTAMENTO?

Contar departamentos de quienes se fueron (filtro empleados que se fueron y miro qué departamentos aparecen más)

In [ ]:
df[df["Attrition"] == "Yes"]["Department"].value_counts().reset_index()

,Department,count
0,Research & Development,126
1,Sales,89
2,Human Resources,13


**Interpretación**
- El departamento de ventas es el que más rotación tiene, seguido de investigación y desarrollo y luego recursos humanos.
Esto puede indicar que los empleados de ventas tienen más probabilidades de irse, lo que podría ser un área de preocupación para la empresa.

### PREGUNTA3. ¿Qué puestos tienen más riesgo de rotación?  JobRole, Attrition  Identificar roles con mayor salida. 
 
- Ver todos los puestos con más rotación, para saber cuales tiene más tasa de abandono:

In [ ]:
#PREGUNTA3. QUE PUESTOS TIENE MAS RIESGO DE ROTACION?
 
#Ver todos los puestos con más rotación, para saber cuales tiene mas tasa d abandono:
 
df[df["Attrition"] == "Yes"]["JobRole"].value_counts().reset_index()  #calcula por volumen y no por tasa, puede llevar a conclusiones erroneas

,JobRole,count
0,lABORATORY tECHNICIAN,62
1,sALES eXECUTIVE,57
2,rESEARCH sCIENTIST,47
3,sALES rEPRESENTATIVE,33
4,hUMAN rESOURCES,13
5,mANUFACTURING dIRECTOR,10
6,hEALTHCARE rEPRESENTATIVE,9
7,mANAGER,5
8,rESEARCH dIRECTOR,2


In [ ]:
#para calcular la tasa de rotación por puesto, necesitamos el número de empleados que se fueron por cada puesto (absolutos) y el número total de empleados por cada puesto (total).
absolutos = df[df["Attrition"] == "Yes"]["JobRole"].value_counts().reset_index()
absolutos.columns = ["JobRole", "Bajas"]

# Total de empleados por puesto
total = df["JobRole"].value_counts().reset_index()
total.columns = ["JobRole", "Total"]

# Unir y calcular tasa de rotación
rotacion = absolutos.merge(total, on="JobRole")
rotacion["Tasa_%"] = (rotacion["Bajas"] / rotacion["Total"] * 100).round(1)
rotacion = rotacion.sort_values("Tasa_%", ascending=False)

rotacion

,JobRole,Bajas,Total,Tasa_%
3,sALES rEPRESENTATIVE,33,83,39.80
4,hUMAN rESOURCES,13,53,24.50
0,lABORATORY tECHNICIAN,62,259,23.90
1,sALES eXECUTIVE,57,327,17.40
2,rESEARCH sCIENTIST,47,293,16.00
6,hEALTHCARE rEPRESENTATIVE,9,131,6.90
5,mANUFACTURING dIRECTOR,10,146,6.80
7,mANAGER,5,102,4.90
8,rESEARCH dIRECTOR,2,80,2.50


**Interpretaciones clave**

- Sales Representative es el puesto más crítico — casi el 40% abandona la empresa. Es un rol orientado al cliente y clave para la expansión europea, lo que lo hace doblemente preocupante.
- Human Resources y Laboratory Technician también superan el 20% de rotación — niveles muy altos que merecen atención.
- Research Director y Manager son los puestos más estables, con tasas por debajo del 5% — perfiles senior con mayor vinculación a la empresa.
- Para el análisis del curso de idiomas: tiene poco sentido invertir en Sales Representative si casi el 40% se va. Habría que cruzarlo con otros filtros.

### PREGUNTA 4: ¿La satisfacción laboral influye en la salida?  JobSatisfaction, EnvironmentSatisfaction, Attrition  Evaluar impacto de la satisfacción. 

In [29]:
# Ver satisfacción de quienes se fueron:
df[df["Attrition"] == "Yes"]["JobSatisfaction"].value_counts().reset_index()

,JobSatisfaction,count
0,3.00,71
1,1.00,64
2,4.00,51
3,2.00,46


In [30]:
#dato en porcentaje:
df[df["Attrition"] == "Yes"]["JobSatisfaction"].value_counts(normalize=True) * 100

JobSatisfaction
3.00   30.60
1.00   27.59
4.00   21.98
2.00   19.83
Name: proportion, dtype: float64

In [31]:
#tenemos 29 nulos:
df["JobSatisfaction"].isnull().sum()

np.int64(29)

In [32]:
# PREGUNTA 4. ¿LA SATISFACCIÓN LABORAL INFLUYE EN LA SALIDA? (combinado con lo anterior, para ver si hay relación entre puesto, satisfacción y rotación)

result = df.groupby("Attrition")["JobSatisfaction"].value_counts(normalize=True).mul(100).round(1).unstack()
result.columns = [f"Nivel {int(c)}" for c in result.columns]

result

,Nivel 1,Nivel 2,Nivel 3,Nivel 4
Attrition,,,,
No,18.30,18.90,29.50,33.30
Yes,27.60,19.80,30.60,22.00


In [26]:
# PREGUNTA 4. ¿LA SATISFACCIÓN LABORAL INFLUYE EN LA SALIDA?

df.groupby("Attrition")[["JobSatisfaction", "EnvironmentSatisfaction"]].mean().round(2)

,JobSatisfaction,EnvironmentSatisfaction
Attrition,,
No,2.78,2.77
Yes,2.47,2.47


**Conclusiones:**
1. Los que se van tienen menor satisfacción en ambas métricas
Tanto en satisfacción laboral como en el entorno, los empleados que abandonan puntúan consistentemente por debajo de los que se quedan.
2. La diferencia no es dramática pero es consistente
~0.3 puntos de diferencia en ambas variables. No es un abismo, pero la consistencia entre las dos métricas refuerza que la satisfacción sí es un factor que influye.
3. Ningún grupo está realmente satisfecho
La escala es del 1 al 4 y los que se quedan solo puntúan 2.78 — esto indica un problema general de satisfacción en la empresa, no solo entre los que se van.

### PREGUNTA 5:¿Las horas extra están relacionadas con mayor rotación? | OverTime, Attrition | Ver si trabajar más horas aumenta la salida. 

In [25]:
# PREGUNTA 5. ¿LAS HORAS EXTRA ESTÁN RELACIONADAS CON MAYOR ROTACIÓN?

overtime = df.groupby("OverTime")["Attrition"].value_counts().unstack(fill_value=0)
overtime["Tasa de Rotación (%)"] = (overtime["Yes"] / (overtime["Yes"] + overtime["No"]) * 100).round(1)
overtime = overtime.reset_index()
overtime.columns = ["Horas Extra", "Se quedaron", "Se fueron", "Tasa de Rotación (%)"]

overtime

,Horas Extra,Se quedaron,Se fueron,Tasa de Rotación (%)
0,No,920,105,10.20
1,Yes,280,125,30.90


**Interpretaciones:**
1. Las horas extra triplican la rotación
Empleados con horas extra tienen una tasa del 30.9% frente al 10.2% de los que no las hacen.
2. Es una de las variables más impactantes del dataset
Una diferencia de 20 puntos porcentuales es muy significativa — pocas variables tendrán tanto peso como esta.
3. Coste oculto para ABC Corporation
Las horas extra pueden parecer productivas a corto plazo, pero si 1 de cada 3 empleados acaba marchándose, el coste real de reemplazarlos supera con creces el beneficio.
4. Conecta con la pregunta anterior
Más horas → menos equilibrio trabajo-vida → menor satisfacción → mayor rotación. Es una cadena de causa-efecto que refuerza los resultados de la pregunta 4.

### PREGUNTA 6: ¿La distancia desde casa afecta a la retención? | DistanceFromHome, Attrition | Analizar si vivir lejos influye en irse. 

In [34]:
# PREGUNTA 6. ¿LA DISTANCIA DESDE CASA AFECTA A LA RETENCIÓN?

df.groupby("Attrition")["DistanceFromHome"].mean().round(2)

Attrition
No     8.91
Yes   10.68
Name: DistanceFromHome, dtype: float64

**Conclusiones:**
1. Los que se van viven más lejos
Una diferencia de casi 2 km de media — los que abandonan viven a 10.68 km frente a los 8.91 de los que se quedan.
2. El impacto existe pero es moderado
No es una diferencia dramática. La distancia influye pero probablemente como factor acumulativo junto a otros como horas extra o baja satisfacción, no por sí sola.
3. La mediana es más reveladora que la media
La mediana de los que se van es 9 km frente a 7 km de los que se quedan — confirma el patrón sin que los valores extremos distorsionen.

### PREGUNTA 7. ¿El salario mensual influye en la rotación? | MonthlyIncome, Attrition | Comparar ingresos entre quienes se quedan y se van. |

In [35]:
# PREGUNTA 7. ¿EL SALARIO MENSUAL INFLUYE EN LA ROTACIÓN?

df.groupby("Attrition")["MonthlyIncome"].describe().round(2)

,count,mean,std,min,25%,50%,75%,max
Attrition,,,,,,,,
No,1226.00,6832.69,4813.70,1051.00,3211.25,5205.00,8836.25,19999.00
Yes,234.00,4741.03,3628.36,1009.00,2366.25,3187.00,5824.25,19859.00


Conclusiones:
1. El salario es una de las variables más impactantes
Los que se van cobran de media 2.091$ menos al mes que los que se quedan. Una diferencia del 30% — muy significativa.
2. La mediana lo confirma sin distorsión
La mediana de los que se van es 3.187$ frente a 5.205$ de los que se quedan. Casi 2.000$ de diferencia — el patrón es claro y consistente.
3. El salario bajo es un factor de riesgo claro
Junto con las horas extra y la baja satisfacción, el salario bajo forma el núcleo de los factores que empujan a los empleados a irse.
4. Implicación para ABC Corporation
Revisar la política salarial, especialmente en los puestos con mayor rotación como Sales Representative y Laboratory Technician, podría tener un impacto directo en la retención.

### PREGUNTA 8: ¿La antigüedad reduce o aumenta la rotación? | YearsAtCompany, TotalWorkingYears, YearsInCurrentRole | Ver cómo influye la experiencia y permanencia. |

### PREGUNTA 9:  ¿La falta de promoción está relacionada con la salida? | YearsSinceLastPromotion, Attrition | Evaluar si no promocionar aumenta la rotación. |

### PREGUNTA 10: ¿Qué variables necesitan limpieza antes del análisis? | EmployeeCount, Over18, StandardHours + nulos/duplicados | Identificar columnas constantes y problemas de calidad. |

## % de mujeres y hombres 

In [13]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
Age,1401.00,36.94,9.11,18.00,30.00,36.00,43.00,60.00
DailyRate,1474.00,802.70,403.54,102.00,465.00,803.00,1157.00,1499.00
DistanceFromHome,1474.00,9.20,8.10,1.00,2.00,7.00,14.00,29.00
Education,1474.00,2.91,1.02,1.00,2.00,3.00,4.00,5.00
EmployeeCount,1474.00,1.00,0.00,1.00,1.00,1.00,1.00,1.00
EmployeeNumber,1474.00,1024.47,602.42,1.00,488.75,1020.50,1555.75,2068.00
EnvironmentSatisfaction,1474.00,2.72,1.09,1.00,2.00,3.00,4.00,4.00
HourlyRate,1474.00,65.89,20.31,30.00,48.00,66.00,83.75,100.00
JobInvolvement,1474.00,2.73,0.71,1.00,2.00,3.00,3.00,4.00
JobLevel,1474.00,2.06,1.11,1.00,1.00,2.00,3.00,5.00


In [14]:
df["BusinessTravel"].value_counts()

BusinessTravel
Travel_Rarely        955
Travel_Frequently    263
Non-Travel           139
Name: count, dtype: int64

In [15]:
df["Gender"].value_counts()

Gender
Male      885
Female    589
Name: count, dtype: int64

In [16]:
df["Gender"].value_counts(normalize=True) * 100

Gender
Male     60.04
Female   39.96
Name: proportion, dtype: float64

,Department,count
0,Research & Development,126
1,Sales,89
2,Human Resources,13


In [4]:
cond_18 = df['Over18'] == 'Yes'
df18 = df[cond_18]
df18

,Age,Attrition,BusinessTravel,DailyRate,Department,DistanceFromHome,Education,EducationField,EmployeeCount,EmployeeNumber,EnvironmentSatisfaction,Gender,HourlyRate,JobInvolvement,JobLevel,JobRole,JobSatisfaction,MaritalStatus,MonthlyIncome,MonthlyRate,NumCompaniesWorked,Over18,OverTime,PercentSalaryHike,PerformanceRating,RelationshipSatisfaction,StandardHours,StockOptionLevel,TotalWorkingYears,TrainingTimesLastYear,WorkLifeBalance,YearsAtCompany,YearsInCurrentRole,YearsSinceLastPromotion,YearsWithCurrManager


In [ ]:
df['EducationField'].unique()
#quitar los espacios con .strip()

<StringArray>
[   'Life Sciences',            'Other',          'Medical',
                nan,        'Marketing', 'Technical Degree',
  'Human Resources']
Length: 7, dtype: str

In [27]:
df['MaritalStatus'].unique()

<StringArray>
['Single', 'Married', 'Divorced', nan, 'Marreid']
Length: 5, dtype: str

In [ ]:
df['Over18'].unique()
#Solo hay un valor en la columna Over18, por lo tanto, no aporta nada a nuestro análisis, podemos eliminarla.

In [ ]:
df['StandardHours'].unique()
#Solo hay un valor en la columna StandardHours, por lo tanto, no aporta nada a nuestro análisis, podemos eliminarla.

array([80., nan])